src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/GroovyCallSiteSelector.java

```diff
@@ -80,6 +80,24 @@ private static boolean matches(@Nonnull Class<?>[] parameterTypes, @Nonnull Obje
         return true;
     }
 
+    private static boolean isInstancePrimitive(@Nonnull Class<?> type, @Nonnull Object instance) {
+        if (type.isInstance(instance)) {
+            return true;
+        }
+        // https://docs.oracle.com/javase/specs/jls/se8/html/jls-5.html#jls-5.1.2
+        if (instance instanceof Number) {
+            Number n = (Number) instance;
+            if (type == Long.class || type == Double.class) {
+                return true;
```

Comment:
`type` here is the method parameter type. If it is `long` then we can accept an `int` for `instance`. At line 93 we check whether a `long` could be narrowed to a method taking `int`.